network.go

```diff
@@ -99,7 +116,7 @@ type NetworkConfig struct {
 // Endpoint gathers a network pair and its properties.
 type Endpoint struct {
```

Comment:
An unrelated comment. But in lot of functions we pass in NetPair. I feel in all cases we should pass in the Endpoint struct vs the NetPair struct as this has more info which these functions will need.

Reply:
Also not all endpoints have netpairs. Example is SRIOV today and vhost-user/tap in the future. So using endpoint as the struct that gets passed around will ensure that there is no significant rewrite in the future as we add support for more interface types.